conn.go

```diff
@@ -35,6 +35,7 @@ type Conn struct {
 	NetNS    int         // Network namespace netlink will interact with.
 	sync.Mutex
 	messages []netlink.Message
+	rules    []*Rule
```

Comment:
Flush had a clear purpose which was to send netlink messages and receive back the response. Making it aware of rules, makes it less clean. Why just rules? How about sets, should it be added too? I think rule specific logic should be handled by rule specific funcs, sets specific logic  by sets funcs etc.. It will keep logic nice and clean and easy to follow.

Reply:
I'll try to make it more generic